src/forwarddiff.jl

```diff
@@ -60,7 +60,10 @@ function nbinomlogpdf(r::ForwardDiff.Dual{T}, p::Real, k::Int) where {T}
 end
 
 ## ForwardDiff broadcasting support ##
-
-function Distributions.logpdf(d::DiscreteUnivariateDistribution, k::ForwardDiff.Dual)
-    return logpdf(d, convert(Integer, ForwardDiff.value(k)))
+if !DISTRIBUTIONS_HAS_GENERIC_UNIVARIATE_PDF
+    @eval begin
+        function Distributions.logpdf(d::DiscreteUnivariateDistribution, k::ForwardDiff.Dual)
```

Comment:
DISTRIBUTIONS_HAS_GENERIC_UNIVARIATE_PDF just indicates if we use Distributions >= 0.24 or Distributions < 0.24. The main breaking change in Distributions 0.24 is that `logpdf` is defined for inputs of type `Real` which are then converted to the support of the distributions (such as integers) in their concrete implementations. Thus it is not needed to have a special function for dual numbers that performs the conversion (and actually this method leads to method ambiguity errors since even discrete distributions now define `logpdf(::MyDistribution, ::Real)`, see, e.g., https://github.com/JuliaStats/Distributions.jl/blob/ae2d6c5c68abc0aeafdc249ca60df058c19279cb/src/univariate/discrete/binomial.jl#L119).

Reply:
I see. Please this as a comment there.

Reply:
I added a comment.